File: android/app/src/main/kotlin/com/baseapp/android/view/blank/BlankLandingView.kt
package com.baseapp.android.view.blank

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.baseapp.android.view.theme.LocalUiTheme

/**
 * Home surface for the `blank` (no template) pack — the native twin of
 * `web/app/src/View/Blank/BlankPage.tsx`. Client-only packs never deploy a
 * backend, so this view uses no stores, components, or network: just a mark
 * and the invitation to start creating. The web logo asset is not bundled in
 * the Android app, so a Material sparkle icon stands in as the app mark.
 */
@Composable
fun BlankLandingView() {
    val theme = LocalUiTheme.current
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(theme.colors.appBg),
        contentAlignment = Alignment.Center,
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(theme.spacing.lg),
        ) {
            Icon(
                imageVector = Icons.Filled.AutoAwesome,
                contentDescription = null,
                tint = theme.colors.accent,
                modifier = Modifier.size(56.dp),
            )
            Text(
                text = "Chat with the agent to start creating.",
                color = theme.colors.textSecondary,
                fontSize = theme.typography.sizes.lg,
                textAlign = TextAlign.Center,
            )
        }
    }
}

File: ios/App/View/Blank/BlankLandingView.swift
import SwiftUI

/// Home surface for the `blank` (no template) pack — the native twin of the
/// web `BlankPage`. Purely static: blank projects have no backend, so this
/// view must never touch stores, components, or the network. The web page
/// shows the Repobot logo; that asset isn't in the iOS bundle, so an SF
/// Symbol stands in as the app mark.
struct BlankLandingView: View {
  @Environment(\.uiThemeTokens) private var theme

  var body: some View {
    VStack(spacing: theme.spacing.xl) {
      Image(systemName: "sparkles")
        .font(.system(size: 56, weight: .medium))
        .foregroundStyle(theme.colors.accent)

      Text("Chat with the agent to start creating.")
        .font(.system(size: theme.typography.sizes.lg, weight: .medium))
        .foregroundStyle(theme.colors.textSecondary)
        .multilineTextAlignment(.center)
    }
    .padding(.horizontal, 32)
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .background(theme.colors.appBg)
  }
}
